boot-clj: in `post_install`, update Boot lib too.

MerelyAPseudonym opened the pull request:
Without this change, the “Boot App Version” could get ahead of the “Boot
Lib Version”. (Which isn’t *problem* per se but it’s potentially
surprising.) The following shell interaction demonstrates the multiple
versions at play:

```
$ boot -h
Boot App Version: 2.0.0
Boot Lib Version: 2.1.2
Clojure Version:  1.6.0
…
$ brew upgrade boot-clj
…
$ boot -h
Boot App Version: 2.2.0
Boot Lib Version: 2.1.2
Clojure Version:  1.6.0
…
$ boot -u
…
$ boot -h
Boot App Version: 2.2.0
Boot Lib Version: 2.2.0
Clojure Version:  1.7.0
…
```

mikemcquaid commented:
Any ideas how we can fix http://bot.brew.sh/job/Homebrew%20Pull%20Requests/32075/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_boot_clj/? Thanks!

mikemcquaid commented:
And also maybe the `post_install` needs removed as it's trying to write to weird places too: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/32075/version=yosemite/testReport/junit/brew-test-bot/yosemite/install___boot_clj_2_2_0_yosemite_bottle_tar_gz/

MerelyAPseudonym commented:
I just rebased onto `master` and added [an additional commit](https://github.com/MerelyAPseudonym/homebrew/commit/a819030f37df7485dd7ac82a22a867da4a1fd944) to try to fix the errors that showed up before.

At this point, though, I’m open to just adding a reminder in the caveats to run `boot --update`. I guess let’s see how Travis fares.

tdsmith commented:
Travis error was:
```
Exception in thread "main" java.io.FileNotFoundException:
/Users/travis/.boot/cache/lib/2.2.0/aether.uber.jar (No such file or directory)
```

I think this approach isn't likely to work so closing for now; please feel free to open a PR against homebrew/homebrew-core if you find a way to get this working.
